src/java/org/apache/cassandra/index/sasi/analyzer/AbstractAnalyzer.java

```diff
@@ -38,15 +40,23 @@ public void remove()
         throw new UnsupportedOperationException();
     }
 
-    public abstract void init(Map<String, String> options, AbstractType validator);
+    public void validate(Map<String, String> options, ColumnDefinition cd) throws ConfigurationException
+    {
+        if (!isCompatibleWith(cd.type))
+            throw new ConfigurationException(String.format("%s does not support type %s",
+                                                           this.getClass().getSimpleName(),
+                                                           cd.type.asCQL3Type()));
+    }
+
+    public abstract void init(Map<String, String> options, AbstractType<?> validator);
 
     public abstract void reset(ByteBuffer input);
 
     /**
      * Test whether the given validator is compatible with the underlying analyzer.
      *
-     * @param validator
-     * @return
+     * @param validator the validator to test the compatibility with
+     * @return true if the give validator is compatible, false otherwise
```

Comment:
I think that `isCompatibleWith` can be `protected`, since now it's only called from `validate`.